src/Security/Oidc/Endpoint/OidcLoginAuthenticator.php

```diff
@@ -81,13 +82,15 @@ public function authenticate(ServerRequestInterface $request): LtiLaunchRequest
 
             $originalMessage = new LtiMessage($this->parser->parse($oidcRequest->getLtiMessageHint()));
 
-            $deployment = $this->repository->find($originalMessage->getDeploymentId());
+            $registration = $this->repository->find(
+                $originalMessage->getMandatoryClaim(MessageInterface::CLAIM_REGISTRATION_ID)
```

Comment:
Is this claim really mandatory from the Request ?

Reply:
since we , as a platform, initiated the OIDC, and therefore forged ourselves the lti_message_hint (as a JWT), putting the claim as mandatory actually ensure the OIDC auth as to be performed following an OIDC init that WE did (security increased)

Reply:
not tobe misleaded, here it's not the `id_token` of a regular launch, it's the `lti_message_hint` that we own, that should be returned opaque and that we use to find backoriginal launch informations. One of them is the registration id (that we still validate no matter what if intercepted).

Reply:
hope it helps ?